mlflow/pyfunc/__init__.py

```diff
@@ -1675,6 +1677,38 @@ def _validate_function_python_model(python_model):
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
+    if isinstance(python_model, PythonModel):
+        try:
+            init_has_model_param = "model" in inspect.signature(python_model.__init__).parameters
+            init_has_model_assignment = "self.model =" in inspect.getsource(python_model.__init__)
```

Comment:
/ai We want to detect assignment to the `model` attribute. I think we can use use `setter`. Let me know if there is any other approach. Would appreciate example.

Reply:
/ai How can I the assignment in the constructor `__init__`? For example:

```python
class MyModel(PythonModel):
    def __init__(self, model):
        self.model = model <- this assignment
```